lib/saml_idp/signed_info_builder.rb

```diff
@@ -64,21 +65,19 @@ def clean_algorithm_name
     end
     private :clean_algorithm_name
 
-    def secret_key
-      SamlIdp.config.secret_key
+    def encoded
+      key = OpenSSL::PKey::RSA.new(@secret_key, password)
+      Base64.strict_encode64(key.sign(algorithm.new, raw))
     end
-    private :secret_key
+    private :encoded
 
-    def password
-      SamlIdp.config.password
+    def using_config_secret_key?
+      @secret_key == SamlIdp.config.secret_key
     end
-    private :password
 
-    def encoded
-      key = OpenSSL::PKey::RSA.new(secret_key, password)
-      Base64.strict_encode64(key.sign(algorithm.new, raw))
+    def password
+      SamlIdp.config.password unless using_config_secret_key?
```

Comment:
Is this the right logic? Before you had `return unless using_config_secret_key?`

Reply:
Yikes, yep. Good catch.

Reply:
Fixed

Reply:
Should we add a spec to catch this?

Reply:
The test suite has no coverage for any case where a passphrase is used 🙃. I added a test to the SAML response spec so that should cover most of the code paths that are affected by the presence of a passphrase.

Also, I amended the commit so I wouldn't dismiss the review with a rebase.